pkg/cableengine/cableengine.go

```diff
@@ -67,16 +68,25 @@ type engine struct {
 	natEndpointInfoCh   chan *natdiscovery.NATEndpointInfo
 	natDiscoveryPending map[string]int
 	installedCables     map[string]metav1.Time
+	gatewayPod          pod.GatewayPodInterface
 }
 
 // NewEngine creates a new Engine for the local cluster
-func NewEngine(localCluster types.SubmarinerCluster, localEndpoint types.SubmarinerEndpoint) Engine {
+func NewEngine(localCluster types.SubmarinerCluster, localEndpoint types.SubmarinerEndpoint,
+	gwPod pod.GatewayPodInterface) (Engine, error) {
+	if gwPod != nil {
+		if err := gwPod.SetHALabels(v1.HAStatusPassive); err != nil {
+			return nil, err
+		}
+	}
```

Comment:
This seems like an unnecessary coupling, ie having the caller pass an interface for this function to call when the caller could directly call it.  Same in `StartEngine`.

Reply:
Yeah, but we store it to call it on tear down.

Reply:
Yes, but it's not unnecessary.

This allows for better testing. Otherwise if we do it all in main, we can't test the behavior as we do in unit tests.

In fact I think I should probably go further and make the responsibility of declaring itself "active" or "passive" part of the cable driver. So when we have A/A cable drivers they can declare themselves active in multiple places, or not.

I can agree that the initial "passive" labeling can be moved into main. But the "active" probably belongs more inside the engine itself.

Anyway, it's easy to move into main, this is all probably very blurry and subjective, so let me know if you believe it's still better in main.

Reply:
Another solution is to update the pod in the GatewaySyncer - that already polls and updates the status in the Gateway object. Yet another way is to add a simple Gateway watcher that updates the pod label.

Reply:
That makes sense to me, I'll update to handle it from the gateway syncer and see how it looks.